components/automatic-grid/style.css

```diff
@@ -0,0 +1,39 @@
+.automatic-grid-cell {
+    height: 50px;
+}
+
+.guic-col-2_4 {
+    flex: 1 1 20%;
+    max-width: 20%;
+}
+
+.guic-col-1_7 {
+    flex: 1 1 14.285714%;
+    max-width: 14.285714%;
+}
+
+.guic-col-1_5 {
+    flex: 1 1 12.5%;
+    max-width: 12.5%;
+}
+
+.guic-col-1_3 {
+    flex: 1 1 11.111111%;
+    max-width: 11.111111%;
+}
+
+.guic-col-1_2 {
+    flex: 1 1 10%;
+    max-width: 10%;
+}
+
+.guic-col-1_1{
+    flex: 1 1 9.090909%;
+    max-width: 9.090909%;
+}
+
```

Comment:
Are these all variations - 1_2, 1_5? What do these numbers represent? Can you please explain in a comment?

Reply:
In the traditional grid representations, columns are factors of 12 - 1, 2, 3, 4, 6, 12. Where for example if we want to have 3 columns we'll do guic-col-(12/3 = 4). Since however 5, 7, 8, 9, 10, 11 are not factors of 12 and 12 divided by them provide a fraction, I made new classes for the columns. The reason for these numbers is that - 12 / 8 = 1.5 hence guic-col-1_5, 12/10 = 1.2 - guic-col-1_2 and etc. I made it this way so I can divide 12 by the number of columns and use it in the class immediately and use the grid component styles.